bluepysnap/nodes.py

```diff
@@ -217,18 +214,6 @@ def load_population_data(self, population):
         return result
 
 
-# TODO: move to `libsonata` library
-def _complex_query(prop, query):
-    # pylint: disable=assignment-from-no-return
-    result = np.full(len(prop), True)
-    for key, value in query.items():
-        if key == '$regex':
```

Comment:
$regex node have been dropped, only `regex:` prefix is kept.

Reply:
No this should be the opposite. `regex:` should be dropped. This is a bad idea to have the function in the argument. See basically my comment here : 
https://github.com/BlueBrain/libsonata/issues/119#issuecomment-769052911

Reply:
I don't see any definition of `$regex` in https://github.com/AllenInstitute/sonata/blob/master/docs/SONATA_DEVELOPER_GUIDE.md#node-sets-file. I agree with your point. How can we be sure that this would be the only way to interpret? For example what about `{'$regex': {Cell.MTYPE: '.*BP'} }`? This way we can have multiple of regex fields like `{'$regex': {Cell.MTYPE: '.*BP', Cell.ETYPE: '.*bap'} }`. What are your thoughts?

Reply:
The queries are not part of the sonata specs at all. This is something we need to introduce as part of the bluepysnap extra features.
The scope of the `$or` / `$and` differs from the normal scope of the queries. They are "queries operators" as they act on queries themselves and not on a field. Like a "$not" operator can do also.
All the other operators are defined the same way : {the field : {the operator: the parameters}}. With some implicit operators for the most common ones like the equal : 
{"property": "the value to match"}

So to me : 
`{'$regex': {Cell.MTYPE: '.*BP', Cell.ETYPE: '.*bap'} }` should be avoided and we should use:
```
 {$and: [{Cell.MTYPE: `{'$regex':  '.*BP}}, {Cell.ETYPE: {'$regex':'.*bap'}}]
```
With this everything is well defined. And you can easily combine the queries in a homogeneous way : 
```
 {$and: [{Cell.MTYPE: `{'$regex':  '.*BP}}, {Cell.ETYPE: {'$regex':'.*bap'}}, {Node.LAYER: [2, 3]}, {Node.X: {"$lt": 2}}]}
```
Also, you can easily do something like : 
```
query_1 = {Cell.MTYPE: `{'$regex':  '.*BP}}
query_2 = {Cell.ETYPE: {'$regex':'.*bap'}
query_3 = {Node.LAYER: [2, 3]}
query_4 = {Node.X: {"$lt": 2}}
full = {$and: [query_1, query_2,query_3, query_4]}
```
and you have only one way of doing it no matter the operator used for each parts.

What I mean is, if you have :
```
query_1 = {Cell.MTYPE: `{'$regex':  '.*BP}}
query_2 = {Cell.ETYPE: {'$regex':'.*bap'}
```
or 
```
query_1 = {Cell.MTYPE: `{'$regex':  '.*BP}}
query_2 = {Cell.LAYER: {'$gt':4}
```
the combination will always be : 
```
full = {$and: [query_1, query_2]}
```